### ISSUE TITLE
Not sure if I'm using gui.update() correctly

### ISSUE THREAD

**Issue by Code12x** [stefvanschie/IF#236]
tl;dr: My gui isn't updating in real time when I call gui.update()

Hey, I'm working on a gui that has a list of items in it. When you click one of the items in the gui, it will move all the items before it's position to the right one, and put the clicked item at the very beginning of the gui. It does work, however, I must manually close the gui and reopen it to see the items in the rearranged positions. I've looked over the information about the gui.update() in the wiki, but I'm wondering if there's something I'm missing, because it's not updating the gui in real time.

Here's my code in question (I have another class that calls the gui.show(player)):
```
public CustomizeOrderGui(int rows, List<String> menuGroupList){
        gui = new ChestGui(rows, "Customize the Order");
        gui.setOnGlobalClick(event -> event.setCancelled(true));

        StaticPane menuGroupListPane = new StaticPane(0, 0, 9, rows);

        for(String menuGroup : menuGroupList){
            String displayName = menuGroup + " (" + data.getConfig().getString("menuGroupID." + menuGroup + ".displayName") + ")";
            Material material = data.getConfig().getItemStack("menuGroupID." + menuGroup + ".item").getType();
            ItemStack menuGroupItemStack = new ProcessItemStack().setMaterial(material).setDisplayName(displayName).getItemStack();

            int menuGroupPos = data.getConfig().getInt("menuGroupID." + menuGroup + ".position");

            GuiItem menuGroupGuiItem = new GuiItem(menuGroupItemStack, event -> {
                for(String menuGroup1 : menuGroupList) {
                    int menuGroup1CurrentPos = data.getConfig().getInt("menuGroupID." + menuGroup1 + ".position");
                    if(menuGroup1CurrentPos < menuGroupPos) {
                        data.getConfig().set("menuGroupID." + menuGroup1 + ".position", menuGroup1CurrentPos + 1);
                    }
                }
                data.getConfig().set("menuGroupID." + menuGroup + ".position", 0);

                data.saveConfig();

                gui.update();
            });
            List<Integer> cords = GuiMath.cordsFromPosition(menuGroupPos, 9);

            menuGroupListPane.addItem(menuGroupGuiItem, cords.get(0), cords.get(1));
        }
        gui.addPane(menuGroupListPane);
    }
```

I did add the [drawing example gui](https://github.com/stefvanschie/IF/wiki/Drawing) in my plugin to make sure that the gui.update is working. And it worked as expected. So that was leading me to wonder if it has something to do with there being a loop before the update is called. But I really don't know.

Any insight would be greatly appreciated.

Thank you

**Comment by Code12x** [comment]
Hello, it seems a fresh look at the code helped me solve the problem. I don't think I was using the gui.update() correctly, but I fixed it by replacing the gui.update(); line with:

```
CustomizeOrderGui customizeOrderGui = new CustomizeOrderGui(rows, menuGroupList);
customizeOrderGui.getGui().show(event.getWhoClicked());
```

Anyways, thanks for this amazing framework!
